quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java

```diff
@@ -110,6 +110,9 @@ private DefaultCacheManager startCacheManager() {
         if (metricsEnabled) {
             builder.getGlobalConfigurationBuilder().addModule(MicrometerMeterRegisterConfigurationBuilder.class);
             builder.getGlobalConfigurationBuilder().module(MicrometerMeterRegisterConfigurationBuilder.class).meterRegistry(Metrics.globalRegistry);
+            builder.getGlobalConfigurationBuilder().cacheContainer().statistics(true);
+            builder.getGlobalConfigurationBuilder().metrics().namesAsTags(true);
```

Comment:
Just a suggestion, add an SPI option to enable histograms so Infnispan reports latencies in `Timer` meters (https://docs.micrometer.io/micrometer/reference/concepts/timers.html).
Since it has a higher impact on performance (last time we checked years ago), I would suggest having it disabled by default.